### PR TITLE
Update EIP-5216 with address zero control in _approve function like EIP-20 

### PR DESCRIPTION
**Pull request by ivanmmurciaua**
After discussing _approve function requires here : https://ethereum-magicians.org/t/last-call-eip-5216-erc1155-approval-by-amount/9898/3 , I'm just updating it

### CODE REVIEW

File: assets/eip-5216/ERC1155ApprovalByAmount.sol [hunk 1]
```diff
@@ -108,7 +110,9 @@ abstract contract ERC1155ApprovalByAmount is ERC1155, IERC1155ApprovalByAmount {
 
         require(idsLength == amountsLength, "ERC1155ApprovalByAmount: ids and amounts length mismatch");
         for (uint256 i = 0; i < idsLength;) {
-            require(allowance(from, to, ids[i]) >= amounts[i], "ERC1155ApprovalByAmount: operator is not approved for that id or amount");
+            if(allowance(from, to, ids[i] < amounts[i])) {
```
Comment (xinbenlv) [inline assets/eip-5216/ERC1155ApprovalByAmount.sol:113]: What's the rationale to not use `_allowances` instead of using `allowances()`?

Comment (ivanmmurciaua) [reply assets/eip-5216/ERC1155ApprovalByAmount.sol:113]: Mmm that's true, thanks for feedback. Also that `if` won't work fine haha. I'll fix and I'll include new commit!

Comment (ivanmmurciaua) [reply assets/eip-5216/ERC1155ApprovalByAmount.sol]: Done @xinbenlv . Thank you very much for your feedback!

**Comment by ivanmmurciaua** [comment]
I think that this EIP would be ready to `final` status. Also the deadline is outdated :)

**Comment by ivanmmurciaua** [comment]
Ready for a review!

**Comment by ivanmmurciaua** [comment]
Yes, review it

**Pull request closed without merging**